lib/firmware.ex

```diff
@@ -90,8 +90,15 @@ defmodule Nerves.Firmware do
   Returns {:error, :await_restart} if the upgrade is requested after
   already updating an image without a reboot in-between.
   """
-  @spec apply(String.t, atom, args \\ []) :: :ok | {:error, reason}
-  def apply(firmware, action, args) do
+  @spec apply(String.t, atom, []) :: :ok | {:error, reason}
+  def apply(firmware, action, args \\ []) do
+    pub_key_path = Application.get_env(:nerves_firmware, :pub_key_path)
+    args = if pub_key_path do
+      IO.puts "using signature!"
```

Comment:
I would avoid `IO.puts` as stdout at this point isn't something useful for most devices, or may actually interfere with intended stdout.   Since we already require Logger, I'd suggest something like
```elixir
Logger.info "#{__MODULE__} using signature"
```